src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1205,7 +1205,9 @@ internal void HandleAltSvc(IEnumerable<string> altSvcHeaderValues, TimeSpan? res
 
                 if (!nextAuthorityPersist)
                 {
+#if !ILLUMOS && !SOLARIS
```

Comment:
I'm not sure if we need to drag it here since neither OS is really supported. 
cc: @am11

Reply:
If I understand it correctly, we were previously using UnknownUnix.cs pattern and not anymore (in favor of supported / unsupporter platform attributes). The compat tool is indicating that this call shouldn't be compiled on unknown unix.

Reply:
@ericstj, was this auto-generated? I know that we define `TARGET_{OS}` constants for OS specific projects, but I am not sure if we define `{OS}`.
https://github.com/dotnet/runtime/blob/e109bbdae8cf66c4ca3bea7df90490a54dfc9f53/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L30-L44

Reply:
Those are added by the SDK here: https://github.com/dotnet/sdk/blob/997376c6bc798af22633a8eb36074a0acec8a200/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L201

Reply:
System.Private.CoreLib doesn't use the TargetPlatform feature which requires them to specify their platform defines themselves.

Reply:
> The compat tool is indicating that this call shouldn't be compiled on unknown unix.

The compiler is actually flagging the illumos and solaris builds of S.N.Http as calling unsupported API.  I chose these defines based on what I saw being passed into the compiler.  I had to ifdef to avoid compiler warnings as error.

@am11 @wfurt can you please let me know if you'd like anything different here?  I'm trying to keep this change as minimal as possible.  I could just pragma suppress the compiler warnings.  I chose not to do that since these APIs were actually throwing on the unsupported platforms.

Reply:
Minimal change looks good to me. There are only a few places which require such conditions, the rest of places are filtered out via Supported/Unsupported OS attributes.

Reply:
I think it is ok. I wish we did not have to drag #ifs to the HTTP code but I understand the reasoning. It would be nice if the tool can simply ignore unfinished/incomplete targets like Solaris of FreeBSD.

Reply:
I was imagining a slightly larger refactor with partial files might also work here but I didn't want to make the diff that large.